optimum/habana/transformers/models/gptj/modeling_gptj.py

```diff
@@ -196,202 +333,226 @@ def forward(
-        embed_positions = embed_positions.to(position_ids.device)
-    repeated_position_ids = position_ids.unsqueeze(-1).repeat(1, 1, embed_positions.shape[-1])
-    sincos = torch.gather(embed_positions, 1, repeated_position_ids)
-    sin, cos = torch.split(sincos, sincos.shape[-1] // 2, dim=-1)
-    sin = sin.contiguous()
-    cos = cos.contiguous()
-
-    for i, (block, layer_past) in enumerate(zip(self.h, past_key_values)):
-        # Model parallel
-        if self.model_parallel:
-            torch.cuda.set_device(hidden_states.device)
-            # Ensure layer_past is on same device as hidden_states (might not be correct)
-            if layer_past is not None:
-                layer_past = tuple(past_state.to(hidden_states.device) for past_state in layer_past)
-            # Ensure that attention_mask is always on the same device as hidden_states
-            if attention_mask is not None:
-                attention_mask = attention_mask.to(hidden_states.device)
-            if isinstance(head_mask, torch.Tensor):
-                head_mask = head_mask.to(hidden_states.device)
-        if output_hidden_states:
-            all_hidden_states = all_hidden_states + (hidden_states,)
 
-        if self.gradient_checkpointing and self.training:
-            outputs = self._gradient_checkpointing_func(
-                block.__call__,
-                hidden_states,
-                None,
-                attention_mask,
-                position_ids,
-                head_mask[i],
-                use_cache,
-                output_attentions,
-                None,
-                sin,
-                cos,
-            )
+    def allocate_kv_cache(self, batch_size, max_seq_len, inp_seq_len):
+        for layer in self.h:
+            layer.allocate_kv_cache(batch_size, max_seq_len, inp_seq_len)
+
+    def reorder_kv_cache(self, beam_idx: torch.LongTensor):
+        return tuple(layer.reorder_kv_cache(beam_idx) for layer in self.h)
+
+    def update_sincos_cache(self, seq_len):
+        for layer in self.h:
+            layer.update_sincos_cache(seq_len)
+
+    def forward(
+        self,
+        input_ids: Optional[torch.LongTensor] = None,
+        past_key_values: Optional[Tuple[Tuple[torch.Tensor]]] = None,
+        attention_mask: Optional[torch.FloatTensor] = None,
+        token_type_ids: Optional[torch.LongTensor] = None,
+        position_ids: Optional[torch.LongTensor] = None,
+        head_mask: Optional[torch.FloatTensor] = None,
+        inputs_embeds: Optional[torch.FloatTensor] = None,
+        use_cache: Optional[bool] = None,
+        output_attentions: Optional[bool] = None,
+        output_hidden_states: Optional[bool] = None,
+        return_dict: Optional[bool] = None,
+        token_idx: Optional[torch.Tensor] = None,
+        reuse_cache: Optional[bool] = False,
+        cache_idx: Optional[int] = None,
+        sin: Optional[torch.Tensor] = None,
+        cos: Optional[torch.Tensor] = None,
```

Comment:
```suggestion
```
I am a bit confused! which upper level these args are coming from. You are overwriting them later

Reply:
These are original code.

Reply:
I do not see any `sin` and `cos` in the original code's forward interface. Can you post the link to that. Even if it is the case st is not right!

Reply:
do you mean this?
https://github.com/huggingface/optimum-habana/blob/main/optimum/habana/transformers/models/gptj/modeling_gptj.py#L214

Reply:
Oh, thanks, you got this code from optimum-habana not transformers. I do not see what is the point of having this argument in the interface unless there is justification for that or you correct the logic inside? 

@regisss your input will help here

Reply:
Yes, I agree it looks strange. It comes from https://github.com/huggingface/optimum-habana/pull/468.
@BaihuiJin Do you remember why you added that?

Reply:
Oh, I was meant to pass these arguments to the `gaudi_gptj_block_forward`, as previously sin and cos was generated in in each blocks, which could be optimized. In the first design I tried to acquire sin and cos in the upper level and pass to `forward`, hence the arguments. But in the final design I changed to generate sincos in `forward` method. I must have forgotten to remove these arguments. It's a mistake, sorry. I think its ok to remove them.

Reply:
removed these.